### PR TITLE
Use SBG sensor status

### PR DESCRIPTION
**Pull request by jonapap**

### CODE REVIEW

**Review by NoahSprenger — COMMENTED** [review]
Looking good so far!

File: libraries/messages/src/sensor_status.rs
Comment (seofernando25) [inline libraries/messages/src/sensor_status.rs:1]: wow good work with this file haha

**Review by seofernando25 — DISMISSED** [review]
lgtm :+1: 
@NoahSprenger  don't forget the clone thingy on the sender loop

File: libraries/messages/src/logging/log.rs [hunk 1]
```diff
@@ -5,30 +5,30 @@ use serde::{Deserialize, Serialize};
...
 use ts_rs::TS;
 
-#[cfg(test)]
+#[cfg(any(feature = "std", test))]
 use proptest_derive::Arbitrary;
 
 #[derive(Serialize, Deserialize, Clone, Debug, Format)]
+#[cfg_attr(any(feature = "std", test), derive(Arbitrary))]
 #[cfg_attr(feature = "ts", derive(TS))]
 #[cfg_attr(feature = "ts", ts(export))]
```
[the remark this answers is not in the document]
Comment (jonapap) [reply libraries/messages/src/logging/log.rs]: Just pushed a commit for this

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -46,6 +46,22 @@ jobs:
 #        with:
 #          command: make
 #          args: test-device --no-run
+  test:
+    runs-on: ubuntu-latest
+    env:
+      RUST_MIN_STACK: 8388608
```
[the remark this answers is not in the document]
Comment (NoahSprenger) [reply .github/workflows/build.yml:52]: Unfortunately, it is needed as I tested before with the stack increase only in the Cargo Make.

File: Makefile.toml [hunk 3]
```diff
@@ -22,7 +22,8 @@ dependencies = [
 
 [tasks.test-messages]
 command = "cargo"
-args = ["test", "-p", "messages", "--target", "${CARGO_MAKE_RUST_TARGET_TRIPLE}"]
+args = ["test", "-p", "messages", "--target", "${CARGO_MAKE_RUST_TARGET_TRIPLE}", "--features", "std", "--no-default-features"]
+env = {RUST_MIN_STACK = "8388608"}
```
[the remark this answers is not in the document]
Comment (NoahSprenger) [reply Makefile.toml:26]: :+1: The value is not special.

File: libraries/messages/src/sensor_status.rs [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+        const ZuptUsed				= (0x00000001 << 26);
+        const AlignValid			= (0x00000001 << 27);
+        const DepthUsed			    = (0x00000001 << 28);
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Format, Copy)]
+#[cfg_attr(any(feature = "std", test), derive(Arbitrary))]
+#[cfg_attr(feature = "ts", derive(TS))]
+#[cfg_attr(feature = "ts", ts(export))]
```
[the remark this answers is not in the document]
Comment (NoahSprenger) [reply libraries/messages/src/sensor_status.rs]: @seofernando25 I thought we still used these bindings. May you please confirm :)

File: libraries/sbg-rs/src/data_conversion.rs [hunk 5]
```diff
@@ -1,123 +1,183 @@
...
+
         Self {
-            time_stamp: value.timeStamp,
-            status: value.status,
-            pressure_abs: value.pressureAbs,
-            altitude: value.altitude,
-            pressure_diff: value.pressureDiff,
-            true_airspeed: value.trueAirspeed,
-            air_temperature: value.airTemperature,
+            time_stamp: value.timeStamp, // TODO: check if valid.
```
[the remark this answers is not in the document]
Comment (NoahSprenger) [reply libraries/sbg-rs/src/data_conversion.rs:83]: Not critical. This timestamp is purely the SBG lifetime timestamp. Further there didn't appear a way to check if it was valid as the bit mask has no corresponding value to validate this field.

**Pull request merged**